src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -1,43 +1,217 @@
+#nullable enable
+
 using System.Net;
 using System.Net.Http.Headers;
-using GraphQL.Instrumentation;
+using GraphQL.Server.Transports.AspNetCore.Errors;
 using GraphQL.Transport;
 using GraphQL.Types;
+using GraphQL.Validation;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace GraphQL.Server.Transports.AspNetCore;
 
-/// <summary>
-/// ASP.NET Core middleware for processing GraphQL requests. Can processes both single and batch requests.
-/// See <see href="https://www.apollographql.com/blog/query-batching-in-apollo-63acfd859862/">Transport-level batching</see>
-/// for more information. This middleware useful with and without ASP.NET Core routing.
-/// <br/><br/>
-/// GraphQL over HTTP <see href="https://github.com/APIs-guru/graphql-over-http">spec</see> says:
-/// GET requests can be used for executing ONLY queries. If the values of query and operationName indicates that
-/// a non-query operation is to be executed, the server should immediately respond with an error status code, and
-/// halt execution.
-/// <br/><br/>
-/// Attention! The current implementation does not impose such a restriction and allows mutations in GET requests.
-/// </summary>
-/// <typeparam name="TSchema">Type of GraphQL schema that is used to validate and process requests.</typeparam>
-public class GraphQLHttpMiddleware<TSchema> : IMiddleware
+/// <inheritdoc/>
+/// <typeparam name="TSchema">
+/// Type of GraphQL schema that is used to validate and process requests.
+/// Specifying <see cref="ISchema"/> will pull the registered default schema.
```

Comment:
```suggestion
/// This may be typed schema as well as <see cref="ISchema"/>. In both cases registered schemas will be
/// pulled from dependency injection framework. Note that when specifying <see cref="ISchema"/> the
/// first schema registered via AddSchema will be pulled (so called default schema).
```

Reply:
```suggestion
/// This may be a typed schema as well as <see cref="ISchema"/>.  In both cases registered schemas will be pulled from
/// the dependency injection framework.  Note that when specifying <see cref="ISchema"/> the first schema registered via
/// <see cref="GraphQL.GraphQLBuilderExtensions.AddSchema{TSchema}(IGraphQLBuilder, TSchema)">AddSchema</see>
/// will be pulled (the "default" schema).
```

Reply:
Revised with a couple grammatical revisions.